automation/taskcluster/decision_task.py

```diff
@@ -104,58 +108,93 @@ def desktop_win32_x86_64_libs():
-        .create()
+def gradle_module_task_name(module, gradle_task_name):
+    return ":%s:%s" % (module, gradle_task_name)
+
+def gradle_module_task(libs_tasks, module_info, is_release):
+    module = module_info['name']
+    if is_release:
+        task_title = "{} - Build, test and upload to bintray".format(module)
+    else:
+        task_title = "{} - Build and test".format(module)
+    task = (
+        android_task(task_title, libs_tasks)
+        .with_script("""
+            yes | sdkmanager --update
+            yes | sdkmanager --licenses
+            ./gradlew --no-daemon clean
+            python automation/taskcluster/release/fetch-bintray-api-key.py
+        """)
+        .with_script("./gradlew --no-daemon {}".format(gradle_module_task_name(module, "testDebug")))
+        .with_script("./gradlew --no-daemon {}".format(gradle_module_task_name(module, "assembleRelease")))
+        .with_script("./gradlew --no-daemon {}".format(gradle_module_task_name(module, "publish")))
+        .with_script("./gradlew --no-daemon {}".format(gradle_module_task_name(module, "zipMavenArtifacts")))
     )
+    for artifact_info in module_info['artifacts']:
+        task.with_artifacts(artifact_info['artifact'])
+    if is_release:
+        if module_info['uploadSymbols']:
+            task.with_scopes("secrets:get:project/application-services/symbols-token")
+            task.with_script("./automation/upload_android_symbols.sh {}".format(module_info['path']))
+        task.with_scopes("secrets:get:project/application-services/publish")
+        task.with_script('./gradlew --no-daemon {} --debug -PvcsTag="$GIT_SHA"'.format(gradle_module_task_name(module, "bintrayUpload")))
+    return task.create()
+
+def build_gradle_modules_tasks(is_release):
+    libs_tasks = libs_for("android", "desktop_linux", "desktop_macos", "desktop_win32_x86_64")
+    module_build_tasks = {}
+    for module_info in module_definitions():
+        module_build_tasks[module_info['name']] = gradle_module_task(libs_tasks, module_info, is_release)
+    return module_build_tasks
+
+def android_multiarch():
+    build_gradle_modules_tasks(False)
 
 def android_multiarch_release():
+    module_build_tasks = build_gradle_modules_tasks(True)
     return (
-        android_task("Android (all architectures): build and test and release")
-        .with_script(TEST_AND_ASSEMBLE_SCRIPT)
+        linux_build_task("All modules - Publish via bintray")
+        .with_dependencies(*module_build_tasks.values())
+        # Since we manipulate secrets, disable logging for a little bit.
         .with_script("""
-            ./automation/upload_android_symbols.sh
             python automation/taskcluster/release/fetch-bintray-api-key.py
-            ./gradlew bintrayUpload --debug -PvcsTag="${GIT_SHA}"
-        """)
-        .with_artifacts("/build/repo/build/target.maven.zip")
-        .with_scopes("secrets:get:project/application-services/symbols-token")
+            set +x
```

Comment:
This seems all kinds of wrong.  Why are we driving bintray with `curl` and not with the (existing, well-tested) Gradle plugin?

Reply:
Note for readers: we discussed this on Slack and agreed this is fine for now, I have however added a long comment explaining why we do this.